Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM node:6-stretch
+LABEL maintainer="will.price94@gmail.com"
+LABEL version="0.0.1"
+RUN apt-get update -qq && \
+    apt-get install -y python
+# Prevent npm from spamming
+RUN npm config set loglevel=warn && \
+    npm config set progress=false
+WORKDIR /app/
+COPY package.json package-lock.json /app/
```

Comment:
i'm slightly confused - are the COPY sources not relative to workdir?

Reply:
COPY copies from the working directory (on the host) when performing the docker build to the target specified inside the container. Here we add the `package.json` files from the working dir and install the deps, this is a common pattern to improve leverage of the docker layer cache. Changes to the source code won't invalidate the layers containing the deps causing subsequent docker builds to repull all the deps.

Reply:
ok - would "COPY package.json package-lock.json ." have been equivalent then?

Reply:
> The <dest> is an absolute path, or a path relative to WORKDIR, into which the source will be copied inside the destination container.

-- https://docs.docker.com/engine/reference/builder/#copy

Yep, I can change it to that if you like? Bit less duplication.

Reply:
Either way - I generally prefer absolute paths, so it's fine as is.  but with WORKDIR it might be more convenient to do everything relative (the npm commands have to be in a particular directory anyways)